net: ethernet: Add EAPoL ether type support

krish2718 opened the pull request:
Wi-Fi protocol uses EAPoL ether type frames for authentication, so, add
support for that ether type so that they are not dropped.

Though we have NET_ETHERNET_FORWARD_UNRECOGNISED_ETHERTYPE to allow
unknown frames to be passed up the stack, but this might cause
performance penalty.

jukkar commented on subsys/net/l2/ethernet/ethernet.c:
```diff
@@ -233,6 +233,7 @@ static enum net_verdict ethernet_recv(struct net_if *iface,
 	switch (type) {
 	case NET_ETH_PTYPE_IP:
 	case NET_ETH_PTYPE_ARP:
+	case NET_ETH_PTYPE_EAPOL:
```
Shouldn't the family be `AF_UNSPEC` because this has nothing to do with IPv4 address family?

krish2718 replied:
Yes.

Commit:
net: ethernet: Add EAPoL ether type support

Wi-Fi protocol uses EAPoL ether type frames for authentication, so, add
support for that ether type so that they are not dropped.

Though we have NET_ETHERNET_FORWARD_UNRECOGNISED_ETHERTYPE to allow
unknown frames to be passed up the stack, but this might cause
performance penalty.

Signed-off-by: Krishna T <krishna.t@nordicsemi.no>
